src/qibolab/instruments/rfsoc.py

```diff
@@ -53,7 +53,22 @@ def pulse_lo_frequency(pulse: Pulse, qubits: dict[int, Qubit]) -> int:
     return lo_frequency
 
 
-def convert_pulse(pulse: Pulse, qubits: dict[int, Qubit]) -> rfsoc_pulses.Pulse:
+def convert_pulse_sequence(sequence: PulseSequence, qubits: dict[int, Qubit]) -> list[rfsoc_pulses.Pulse]:
+    """Convert PulseSequence to list of rfosc pulses with relative time."""
+
+    abs_time = 0
+    list_sequence = []
+    for pulse in sequence:
+        abs_start = pulse.start * NS_TO_US
+        start_delay = abs_start - abs_time
+        pulse_dict = asdict(convert_pulse(pulse, qubits, start_delay))
+        list_sequence.append(pulse_dict)
+
+        abs_time += start_delay
+    return list_sequence
+
+
+def convert_pulse(pulse: Pulse, qubits: dict[int, Qubit], start_delay: float) -> rfsoc_pulses.Pulse:
```

Comment:
Not very important, but maybe I would put this before `convert_pulse_sequence` because it is used by it.